src/pubsub/ua_pubsub_reader.c

```diff
@@ -786,17 +803,42 @@ UA_DataSetReader *UA_ReaderGroup_findDSRbyId(UA_Server *server, UA_NodeId identi
 /* This callback triggers the collection and reception of NetworkMessages and the
  * contained DataSetMessages. */
 void UA_ReaderGroup_subscribeCallback(UA_Server *server, UA_ReaderGroup *readerGroup) {
+
+    UA_LOG_DEBUG(&server->config.logger, UA_LOGCATEGORY_SERVER, "PubSub subscribe callback");
+
+    if(!readerGroup) {
+        UA_LOG_ERROR(&server->config.logger, UA_LOGCATEGORY_SERVER,
+            "Subscribe failed. ReaderGroup not found");
+        /* TODO: we can't set the state of the specific ReaderGroup to error, because the pointer is invalid ...
+            we would have to set the connection state to error */
```

Comment:
I think this is an edge case because there should be absolutely no reason for this null pointer. The user never gets access to the internal ReaderGroup pointer and therefore I think the error log is enough.

Reply:
ok, I have removed the comment